File: strategy-pattern/js/src/ContextDog.js
import { Context } from './Context.js'
import { StrategyB } from './StrategyB.js'
export class ContextDog extends Context {
  constructor() {
    // 使用某个策略
    super()
    console.log('ContextDog::setStrategy(StrategyB).')
    this.setStrategy(new StrategyB())
  }
}

File: strategy-pattern/python/src/ContextDog.py
# -*- coding: utf-8 -*-
"""
@author: jarry
"""
from src.Context import Context
from src.StrategyB import StrategyB
class ContextDog(Context):
  def __init__(self,):
    print('ContextDog::setStrategy(StrategyB).')
    self.set_strategy(StrategyB())
